#
#  YAML File to be loaded into the ROS Parameter Server
#
#  These parameters support the various controllers.  They are
#  collected under the namespace of the robot.
#
amber3m:

  # The joint_state publisher.  This is a fake controller
  # that causes GAzebo to produce the joint_states topic.
  joint_state_publisher:
    type: joint_state_controller/JointStateController
    publish_rate: 100

  # The position commands.  These controllers simply pass
  # position (angle) commands as values into the simulation.
  joint1_position:
    type: position_controllers/JointPositionController
    joint: LeftKnee
  joint2_position:
    type: position_controllers/JointPositionController
    joint: LeftHip
  joint3_position:
    type: position_controllers/JointPositionController
    joint: RightHip
  joint4_position:
    type: position_controllers/JointPositionController
    joint: RightKnee


  # The effort commands.  These controllers simply pass
  # effort (torque) commands into the simulation.
  joint1_effort:
    type: effort_controllers/JointEffortController
    joint: LeftKnee
  joint2_effort:
    type: effort_controllers/JointEffortController
    joint: LeftHip
  joint3_effort:
    type: effort_controllers/JointEffortController
    joint: RightHip
  joint4_effort:
    type: effort_controllers/JointEffortController
    joint: RightKnee


  # The PD position controllers generating torque commands based on
  # position commands and feedback of position from the simulation.
  # Note this is effort based
  joint1_position_controller:
    type: effort_controllers/JointPositionController
    joint: LeftKnee
    pid: {p: 400.0, i: 0.0, d: 10.0}
  joint2_position_controller:
    type: effort_controllers/JointPositionController
    joint: LeftHip
    pid: {p: 400.0, i: 0.0, d: 10.0}
  joint3_position_controller:
    type: effort_controllers/JointPositionController
    joint: RightHip
    pid: {p: 400.0, i: 0.0, d: 10.0}
  joint4_position_controller:
    type: effort_controllers/JointPositionController
    joint: RightKnee
    pid: {p: 400.0, i: 0.0, d: 10.0}

